src/charm.py

```diff
@@ -110,14 +111,17 @@ def configure(self):
 
     def disable_i40e_lldp(self):
         """Disable i40e."""
-        path = "/sys/kernel/debug/i40e"
-        if not os.path.exists(path):
-            return True
-        for nic in os.listdir(path):
-            cmd = open("{}/{}/command".format(str(path), str(nic)), "w")
-            cmd.write("lldp stop")
-            cmd.close()
+        getnics = "stat --format '%N' /sys/class/net/*/device/driver | grep 'drivers/i40e' | cut -d/ -f 5"
+        nics = subprocess.check_output(getnics, shell=True)
 
+        if nics == "":
+            logger.info("Can't find any i40e NICs")
+            exit(0)
+
+        for nic in nics.decode('UTF-8').splitlines():
+            logger.info("Using ethtool(8) to disable FW lldp for %s" % nic)
+            subprocess.run(['sudo', '/usr/sbin/ethtool', '--set-priv-flags', nic, 'disable-fw-lldp', 'on'])
```

Comment:
What happens when this fails? Is it expected that the user will know what to do in this case? Clearly it needs some operator intervention - but if there are multiple i40e nics where this is disabled it would fail to set for all subsequent ones, where it may be better if it did what it can and then fails only on the ones that error out.

Reply:
We've added an exit check on the ethtool call but shouldn't the charm fail if it fails to set the requested configuration?